lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js

```diff
@@ -35,40 +37,78 @@ class FeatureCard extends React.Component {
   constructor(props) {
     super(props);
 
+    // These variables shouldn't force a state change & subsequent render
+    this.requests = [];
+    this.campaigns = [];
+    this.campaignsToExclude = setting('dsUser.activity.signups', []);
+    this.campaignsApiPage = 1;
+
     this.state = {
-      campaigns: [],
-      current_campaign: '',
-      signup_animation: ''
+      campaignIndex: -1,
+      isSignedUp: ''
     };
   }
 
   componentDidMount() {
-    this.serverRequest = $.get(`${API_URL}campaigns?random=true&staff_pick=true`, result => {
-      // Filter out campaigns users have signed up for & then sort remaining campaigns
-      const userSignups = Drupal.settings.dsUser.activity.signups;
-      const campaigns = result.data.filter(campaign => userSignups.indexOf(parseInt(campaign.id)) == -1).sort(sortCampaigns);
-
+    this.getCampaigns(true, true, 1, () => {
+      // Set the index so the first campaign renders
       this.setState({
-        campaigns: campaigns,
-        current_campaign_index: 0
+        campaignIndex: 0
       });
+
+      if (this.campaigns.length == 0) {
+        // For our super users who already signed up for them all...
+        this.viewAnother();
+      }
+      else {
+        // These are all staff pick campaigns which we don't want to display twice.
+        this.campaignsToExclude = this.campaignsToExclude.concat(this.campaigns.map(campaign => parseInt(campaign.id)));
+      }
     });
   }
 
+  getCampaigns(staffPickOnly, paginate, page, cb) {
```

Comment:
callback it is 👍 ive seen other dev's use cb as a shorthand for it so i got into the habit